### DISCUSSION TITLE
Prest configuring SSL and using HTTPS

### DISCUSSION THREAD

**Discussion by aletik** [prest/prest#712] [API Server]
Hi Guys

Wanted to thank you for the very cool prest library. I like it a lot. Works well over http
I have a question..

I’m using it internally and tried prest http calls from Confluence. Unfortunately our Confluence runs behind https and making any javascript/jquery http call is blocked. I tried configuring SSL in prest.toml for localhost using openssl instructions I saw here (generated cert and key)
https://www.section.io/engineering-education/how-to-get-ssl-https-for-localhost/

but getting the prest SSL not enabled on the server error all the time without really any good details..

Wonder if you can publish instructions on how to make prest run on https on say localhost using openssl on mac?

Thank you very much!!

**Comment by avelino** [comment]
Solution is to put the ssl certificate in the http proxy (example **nginx** or **caddy**), I don't recommend leaving prestd responding directly on the http port without a proxy in front of production environment.

I will open an issue to write a page in the documentation about how to put ssl certificate directly in prestd as you asked

> thanks for the compliment, I personally think prestd is a super cool project that speeds up the development of api rest

**Reply by avelino** [reply]
here #714 
I don't have much time to write this documentation in the next few days, but I will try to prioritize to do it at some point soon.
